private/pkg/git/gitobject/doc.go

```diff
@@ -0,0 +1,17 @@
+// Copyright 2020-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package gitobject implements the core git objects (blobs, commits, trees)
+// as well as a Reader that can read these objects.
+package gitobject
```

Comment:
Same comment, not that you have to change it, but would ask why the separate package? Ie should we smash gitbranch and gitobject into git, and why or why not?